### ISSUE TITLE
Face/Motion detection not working

### ISSUE THREAD

**Issue by aptonline** [thanksmister/android-mqtt-alarm-panel#42]
As per my posts of the HA forums (which seem to be down). Motion detection and face detection isn’t triggering either MQTT topics or waking the screen. Motion and face IS however shown in camera test.

Tested is WallPanel and works as expected.

**Comment by thanksmister** [comment]
That’s strange because they are virtually identical in the code.  So you have the new WallPanel application running in the same device and motion/face work as expected.   They wake the screen and also send the Mqtt messages.  

However if you run the Alarm Panel application then only sensor data is being published but motion/face detection are not working.

It almost sounds like the service isn’t properly reading the setting changes or the camera is throwing an error.  Do you have MJPEG streaming working when motion detection is on?  That’s one big I did find but nothing else yet.

**Comment by thanksmister** [comment]
Didn’t you also have to lower your FPS rate for the video in WallPanel?  I seem to recall us having a similar conversation about WallPanel but not sure.

**Comment by aptonline** [comment]
They are both running on they same device yes.

WallPanel wake via motion and face detection and is sending all the MQTT details , motion, face battery (all the ones my device can do).

With AlarmPanel only batter and state is being sent via MQTT, face and motion not sent or waking. No MPEG streaming even if I turn motion detection on (saw this issue on the ha forums/git hub).

**Comment by thanksmister** [comment]
But what is your FPS setting on WallPanel?   Did you lower it?  There is setting under camera, can you check your WallPanel settings and see if you changed the FPS at all.

**Comment by aptonline** [comment]
Both WallPanel and AlarmPanel are set to 1fps which works for WallPanel in low light. Work on AlarmPanel in camera test but not in use.

**Comment by thanksmister** [comment]
Ok, I will try to see if I can duplicate the issue.  I had one such report with an Amazon Fire 7 Tablet with WallPanel.  Same exact issue that you are reporting.   It could be camera hanging, but then the camera tests in the Settings would also break.  Also you obviously can't run both WallPanel and Alarm Panel at the same time on the same device due to access to the camera.

**Comment by aptonline** [comment]
Yeah I made sure I fully quit WallPanel. I’m reluctant to delete it from my tablet as don’t want to loose the settings but do you think having both on the device could be something to do with it?

**Comment by thanksmister** [comment]
No, you can have both on the same device they just won't play nice if they both run at the same time. The same thing would happen if you used your camera application while also using the camera in either application.  I just want to rule out all possibilities.   But don't delete the app, its fine.

**Comment by aptonline** [comment]
Good news, the recent build with the fix for the MPEG camera steaming issue also seems to have fixed the face and motion detection. MQTT updates for these are also now being sent. I’ll do some further testing to make sure but it’s looking good, thanks.

**Comment by thanksmister** [comment]
Great but I didn’t actually do anything to fix the motion or face detection.  I simply changed MJPEG streaming to work on its own.  Well it’s odd but I’m glad it’s working.

**Comment by aptonline** [comment]
Must hast been something stopping camera access generally.
